pymodels/SI_V25_04/segmented_models.py

```diff
@@ -164,23 +167,33 @@ def dipole_b1(m_accep_fam_name, simplified=False):
     PolyB = _np.zeros(1+max(monomials))
     for i in range(len(segmodel)):
         fam_name, element_type = segtypes[segmodel[i][0]]
+        angle = d2r * segmodel[i][2]
         if element_type == _pyaccel.elements.rbend:
             for j in range(len(monomials)):
                 PolyB[monomials[j]] = segmodel[i][j+3]
             PolyA = 0 * PolyB
             element = element_type(
                 fam_name=fam_name, length=segmodel[i][1],
-                angle=d2r * segmodel[i][2], angle_in=0, angle_out=0,
+                angle=angle, angle_in=0, angle_out=0,
                 gap=0, fint_in=0, fint_out=0,
                 polynom_a=PolyA, polynom_b=PolyB)
         else:
             element = element_type(fam_name)
         model.append(element)
 
+    # --- add source point marker to half-model ---
+    imodel = model[::-1]
+    angles = _np.array([elem.angle for elem in imodel])
+    angles_cumsum = _np.cumsum(angles)
+    src_idx = _np.argmin(_np.abs(angles_cumsum - src_point_angle))
```

Comment:
is this correct? The beamline exit is at 3.2mrad from the start of the field, or from the dipole edge?
How far from 3.2mrad is the result of this `argmin`?

Reply:
@fernandohds564 , inserting the marker this way  it is located after B1 segments with indices 0,1,2,3 and 4, at 3.208 mrad.

Reply:
I committed comments about this.